bridge_adaptivity/module/static/module/js/collection_drag_drop.js

```diff
@@ -1,13 +1,25 @@
-// include base_drag_drop.js before this script
+// WARNING(AndreyLykhoman):include base_drag_drop.js before this script
 
-function before_add_element(element, index) {
+before_add_element = function(){
```

Comment:
why did you change function declaration to function expression?

please add additional spaces between brackets () and {}

Reply:
1) This is a higher order function that has two variables in the closure. There are 'atype' and 'index' variables.  The main anonymous function (I call the function right away) returns a function that set to 'before_add_element' variable. And this makes function expression. 
When I use 'before_add_element' for making an element's label, I can check the 'atype' of elements. If the 'atype' of the element isn't equal than variable 'atype', I will set a zero to the 'index' variable, set a new value to 'atype' variable. 
This allows you to make the correct numbering of the elements after drag & drop.

2) I added additional spaces.